src/main/java/seedu/address/model/event/Time.java

```diff
@@ -45,36 +50,74 @@ public LocalDateTime getTime() {
     /**
      * Returns if a given string is a valid time.
      */
-    // TODO: change this checker whenever more formats are added in the method below
+    // TODO: change this checker whenever more formats are added in the parse method below
     public static boolean isValidTime(String timeInput) {
         try {
-            LocalDateTime.parse(timeInput, DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm"));
+            LocalDateTime.parse(timeInput, DateTimeFormatter.ofPattern(STANDARD_TIME_PATTERN));
             return true;
         } catch (DateTimeParseException e) {
             return false;
         }
     }
 
-    // TODO: add more formats to be parsed here, add them to the valid checking method above
+    // TODO: add more formats to be parsed here, add them to the isValidTime checking method above
     public static LocalDateTime parse(String timeInput) {
-        return LocalDateTime.parse(timeInput, DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm"));
+        return LocalDateTime.parse(timeInput, DateTimeFormatter.ofPattern(STANDARD_TIME_PATTERN));
     }
 
-    // toDisplayString controls the format of time displayed in the GUI panel
+    // getDisplayName controls the format of time displayed in the GUI panel and in the response.
     // TODO: make display prettier/more relevant to the user
-    public String toDisplayString() {
-        return time.format(DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm"));
+    public String getDisplayName() {
+        StringBuilder builder = new StringBuilder();
+
+        builder.append(time.getDayOfWeek().getDisplayName(TextStyle.SHORT_STANDALONE, Locale.ENGLISH) + " ")
+                .append(getDayOfMonthAsString() + " ")
+                .append(time.getMonth().getDisplayName(TextStyle.SHORT_STANDALONE, Locale.ENGLISH) + " ")
+                .append(time.getYear());
+
+        LocalDateTime now = LocalDateTime.now();
+        long dayDifference = DAYS.between(now, time);
+        if (dayDifference < 0) {
+            if (dayDifference < -1) {
+                builder.append(" (" + (dayDifference * -1) + " days ago)");
+            } else {
+                builder.append(" (" + (dayDifference * -1) + " day ago)");
+            }
+        } else if (dayDifference == 0) {
+            builder.append(" (Today)");
+        } else if (dayDifference > 0) {
+            if (dayDifference > 1) {
+                builder.append(" (In " + dayDifference + " days)");
+            } else {
+                builder.append(" (In " + dayDifference + " day)");
+            }
+        }
+
+        return builder.toString();
+    }
+
+    // private method for use in getDisplayName above
+    private String getDayOfMonthAsString() {
+        switch (time.getDayOfMonth()) {
+        case 1:
+            return "1st";
+        case 2:
+            return "2nd";
+        case 3:
+            return "3rd";
+        default:
+            return time.getDayOfMonth() + "th";
```

Comment:
Should add 21st, 22nd, 23rd and 31st as the cases too

Reply:
Ill update with a modulo thing here